### ISSUE TITLE
`assertMatrix` versus `checkMatrix`

### ISSUE THREAD

**Issue by geanders** [rickzhouwen/HDtest#8]
I think that we want `assertMatrix` where we now have `checkMatrix`. This is for two reasons: 

1. If what you're checking is a matrix, `assertMatrix` returns that object _invisible_. On the other hand, `checkMatrix` returns `TRUE`. Ideally, we want quiet evaluation if everything's working okay (so not something that returns a value if things are going like we want, as `checkMatrix` seems to). 
2. If what you're checking is not a matrix, `checkMatrix` seems to only print a message, not cause an error. As a result, when you use it within a helper function, the message about the matrix does not make it to users. I tested `testCov` twice with a non-matrix for `X`, once with `testCovHD` using `checkMatrix` (original code) and once with `testCovHD` using `assertMatrix`. With `checkMatrix`, `testCov` failed with the error message `Error in if (ncol(Y) != p) { : argument is of length zero`, which isn't very helpful. When using `assertMatrix` with an `X` that isn't a matrix, the error message is: 

```
 Error in testCovHD(X = blah, Y = Y, J = 1, alpha = alpha, DNAME = DNAME,  : 
  Assertion on 'X' failed: Must be of type 'matrix', not 'character'. 
```

Before I make the change everywhere, was there some reason why `checkMatrix` was used in the original rather than `assertMatrix`? Any good reason for not changing them all to `assertMatrix`?

**Comment by rickzhouwen** [comment]
I did not write this part in R (I usually do in python or matlab). For some reason the student who did it long time ago just used checkMatrix. Is there an easy way to change it?

**Comment by geanders** [comment]
Yes, this is easy to change, we can just replace every `checkMatrix` with `assertMatrix` from the same package. I just wanted to make sure that there wasn't some important reason for the initial choice to use `checkMatrix` before I made that change.

**Comment by rickzhouwen** [comment]
That sounds great!
